### PR TITLE
Code cleanup and reorganization

### LINKED ISSUE #42: Eliminate `core` package
Now that the task management logic has been moved to its own package, the `core` package contains only data types used by other packages. Once #40 is merged, it's time to dissolve this package and move these types to their rightful homes.

### LINKED ISSUE #43: Rationalize log messages
See @ialarmedalien 's [comment](https://github.com/kbase/dts/pull/40#discussion_r1463573980) in #40.

### LINKED ISSUE #44: Rationalize error types
We've been mostly making do with string-flavored Go error types, but we could do a bit better in the pursuit of more precise error handling. This task represents the assessment of benefits to be gained from a slightly more elaborate error handling protocol. Probably best to do this along with #42 and #43.

### LINKED ISSUE #46: Consolidate test fixtures into `dtstest` package
At the moment, a few of our unit tests make use of test fixtures for (fake) databases and endpoints. Now that we have a `dtstest` package, it makes sense for us to consolidate these fixtures and move them to that package.

### PR DESCRIPTION
**Pull request by jeff-cohere**
This PR is an attempt to clean things up a bit. I doubt it's perfect, but it seems like a structural improvement. Highlights:

* The `core` package has been eliminated, and the `Database` and `Endpoint` interfaces have been moved to the `databases` and `endpoints` packages. Meanwhile, the Frictionless data types have been moved to their own `frictionless` package.
* Some unwinding of dependencies between packages has been performed.
* The separate and almost-morally-equivalent Database and Endpoint test fixtures used by the `tasks` and `services` unit tests have been consolidated in the `dtstest` package. I hope their use makes some more sense than what they replace.
* I've made a cursory attempt to create some meaningful error types for error handling, and I've tried to make log messages a bit more uniform. Probably both are a work in progress.

Closes #42 
Closes #43 
Closes #44 
Closes #46

### CODE REVIEW

File: databases/jdp/database_test.go [hunk 1]
```diff
@@ -110,7 +114,8 @@ func TestResources(t *testing.T) {
 		assert.Equal(jdpSearchResult.Name, resource.Name, "Resource name mismatch")
 		assert.Equal(jdpSearchResult.Path, resource.Path, "Resource path mismatch")
 		assert.Equal(jdpSearchResult.Format, resource.Format, "Resource format mismatch")
-		assert.Equal(jdpSearchResult.Bytes, resource.Bytes, "Resource size mismatch")
+		// FIXME: looks like JDP and JAMO disagree about a resource size!
```
Comment (ialarmedalien) [inline databases/jdp/database_test.go:117]: 😖

File: dtstest/dtstest.go [hunk 2]
```diff
@@ -34,3 +44,230 @@ func EnableDebugLogging() {
...
+func (ep *Endpoint) Status(id uuid.UUID) (endpoints.TransferStatus, error) {
+	if info, found := ep.Xfers[id]; found {
+		if info.Status.Code != endpoints.TransferStatusSucceeded &&
+			time.Now().Sub(info.Time) >= ep.Options.TransferDuration { // update if needed
+			info.Status.Code = endpoints.TransferStatusSucceeded
+			ep.Xfers[id] = info
+		}
+		return info.Status, nil
+	} else {
+		return endpoints.TransferStatus{}, fmt.Errorf("Invalid transfer ID: %s", id.String())
```
Comment (ialarmedalien) [inline dtstest/dtstest.go:178]: optional: no else needed after a return

File: dtstest/dtstest.go [hunk 3]
```diff
@@ -34,3 +44,230 @@ func EnableDebugLogging() {
...
+
+func (db *Database) StagingStatus(id uuid.UUID) (databases.StagingStatus, error) {
+	if info, found := db.Staging[id]; found {
+		endpoint := db.Endpt.(*Endpoint)
+		if time.Now().Sub(info.Time) >= endpoint.Options.StagingDuration { // FIXME: not always so!
+			return databases.StagingStatusSucceeded, nil
+		}
+		return databases.StagingStatusActive, nil
+	} else {
+		return databases.StagingStatusUnknown, nil
```
Comment (ialarmedalien) [inline dtstest/dtstest.go:263]: optional: no else needed after a return

File: endpoints/local/endpoint.go [hunk 4]
```diff
@@ -206,12 +207,12 @@ func (ep *Endpoint) Transfer(dst core.Endpoint, files []core.FileTransfer) (uuid
...
 }
 
-func (ep *Endpoint) Status(id uuid.UUID) (core.TransferStatus, error) {
+func (ep *Endpoint) Status(id uuid.UUID) (endpoints.TransferStatus, error) {
 	if xfer, found := ep.Xfers[id]; found {
 		return xfer.Status, nil
 	} else {
-		return core.TransferStatus{
-			Code: core.TransferStatusUnknown,
+		return endpoints.TransferStatus{
```
Comment (ialarmedalien) [inline endpoints/local/endpoint.go:214]: optional return / else

**Review by ialarmedalien — APPROVED** [review]

File: databases/jdp/database_test.go (hunk 1 above)
Comment (jeff-cohere) [reply databases/jdp/database_test.go:117]: I know... after the end-to-end demo I am going to discuss the JDP/JAMO relationship with Steven W.

**Pull request merged**